### PR TITLE
Auto-increment the Site ID instead of generating a UID.

### PR DESCRIPTION
**Pull request by danielmilner**

### CODE REVIEW

File: js/services.js [hunk 1]
```diff
@@ -228,6 +229,21 @@ angular.module('wpApp.services', [])
 			});
 	}
 	
+	function getNextID() {
+		return $q.when( _db.allDocs( { descending: true, limit: 1 } ) )
```
Comment (shprink) [inline js/services.js]: Is _db.allDocs asynchronous?

Comment (danielmilner) [inline js/services.js]: Yes, the PouchDB API is asynchronous.

Comment (shprink) [inline js/services.js]: Ok if it returns a promise then you do not need `$q.when`, just return `_db.allDocs`

File: js/services.js [hunk 2]
```diff
@@ -184,7 +185,7 @@ angular.module('wpApp.services', [])
 	};
 	
 	function addSite( site ) {
-		return $q.when( _db.post( site ) );
+		return $q.when( _db.put( site ) );
```
Comment (shprink) [inline js/services.js]: same here

File: js/services.js (hunk 1 above)
Comment (danielmilner) [inline js/services.js]: Let me check some things. When doing it that way, I may have been referencing a doc from before PouchDB supported promises.

File: js/services.js [hunk 3]
```diff
@@ -228,6 +229,21 @@ angular.module('wpApp.services', [])
 			});
 	}
 	
+	function getNextID() {
+		return $q.when( _db.allDocs( { descending: true, limit: 1 } ) )
+			.then( function( docs ) {
+				var nextID = 1;
+				if ( docs.total_rows == '1' ) {
+					var nextID = parseInt( docs.rows[0].id );
+					if ( nextID === '' ) {
```
Comment (itsananderson) [inline js/services.js]: parseInt returns `NaN` if the value isn't a valid number. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

`nextID === ''` will always be `false`

A valid check would be `isNaN(nextID)`

Comment (danielmilner) [inline js/services.js]: Thanks. Fixing this.

**Pull request closed without merging**